pkg/runner/healthcheck_common.go

```diff
@@ -0,0 +1,283 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/docker/go-units"
+	"github.com/ipfs/testground/pkg/api"
+	"github.com/ipfs/testground/pkg/docker"
+	"github.com/ipfs/testground/pkg/rpc"
+)
+
+type Checker func() (bool, error)
+type Fixer func() error
+
+// HealthcheckHelper is a strategy interface for runners.
+// Each runner may have required elements -- infrastructure, etc. which should be checked prior to
+// running plans. Individual checks are registered to the HealthcheckHelper using the Enlist()
+// method. With all of the checks enlisted, execute the checks, and optionally fixes, using the
+// RunChecks() method. The details of how the checks are performed is implementation specific.
+// Typically, the checker and fixer passed to the enlist method will be closures. These methods will
+// be called when RunChecks is executed.
+type HealthcheckHelper interface {
+	Enlist(name string, c Checker, f Fixer)
+	RunChecks(ctx context.Context, fix bool) error
+}
```

Comment:
I don't think we'll ever need multiple implementations of the healthchecker interface. If down the pike we decide to parallelise, we can do so without an alternative impl. I'd rm this interface, to simplify.